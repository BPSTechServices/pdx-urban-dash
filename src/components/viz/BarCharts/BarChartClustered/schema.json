{
  "$schema": "https://vega.github.io/schema/vega/v5.json",
  "width": 500,
  "height": 200,
  "padding": 5,

  "data": [
    {
      "name": "table",
      "$dataValues": ""
    }
  ],

  "config": {
    "axis": {
      "$labelColor": "",
      "$labelFontSize": 0
    }
  },

  "signals": [
    {
      "name": "tooltip",
      "value": {},
      "on": [
        {"events": "rect:mouseover", "update": "datum"},
        {"events": "rect:mouseout",  "update": "{}"}
      ]
    }
  ],

  "scales": [
    {
      "name": "xscale",
      "type": "band",
      "domain": {"data": "table", "field": "x"},
      "range": "width",
      "padding": 0.05,
      "round": true
    },
    {
      "name": "yscale",
      "domain": {"data": "table", "field": "y"},
      "range": "height",
      "nice": true,
      "zero": true
    },
    {
      "name": "color",
      "type": "ordinal",
      "$colorRange": [],
      "domain": {"data": "table", "field": "c"}
    },
    {
      "name": "textoffset",
      "type": "band",
      "range": "width",
      "domain": {"data": "table", "field": "c"}
    }
  ],

  "axes": [
    { "orient": "bottom", "scale": "xscale" },
    { "orient": "left", "scale": "yscale" }
  ],

  "marks": [
    {
      "type": "group",
      "from": {
        "facet": {
          "name": "facet",
          "data": "table",
          "groupby": "x"
        }
      },

      "encode": {
        "enter": {
          "x": {"scale": "xscale", "field": "x"}
        }
      },

      "signals": [
        {"name": "width", "update": "bandwidth('xscale')"}
      ],

      "scales": [
        {
          "name": "groupoffset",
          "type": "band",
          "range": "width",
          "domain": {"data": "facet", "field": "c"}
        }
      ],

      "marks": [
        {
          "name": "bars",
          "from": {"data": "facet"},
          "type": "rect",
          "scales": [
            {
              "name": "textoffset",
              "type": "band",
              "range": {"data": "data", "field": "c"},
              "domain": {"data": "facet", "field": "c"}
            }
          ],
          "encode": {
            "enter": {


              "x": {"scale": "groupoffset", "field": "c"},
              "width": {"scale": "groupoffset", "band": 1},

              "y": {"scale": "yscale", "field": "y"},
              "y2": {"scale": "yscale", "value": 0},
              

              "fill": {"scale": "color", "field": "c"}
            },
            "update": {
              "fillOpacity": {"value": 1}
            },
            "hover": {
              "fillOpacity": {"value": 0.5}
            }
          }
        },
        {
          "type": "text",
          "from": {"data": "bars"},
          "encode": {
            "enter": {
              "x": {"field": "x", "offset": {"field": "width", "mult": 0.5}},
              "y": {"field": "y", "offset":-5},
              "fill": {"value": "black"},
              "align": {"value": "center"},
              "text": {"field": "datum.y"}
            },
            "update": {
              "fillOpacity": {"value": 1}
            },
            "hover": {
              "fillOpacity": {"value": 0.5}
            }
          }
        }
      ]
    }
  ]
}