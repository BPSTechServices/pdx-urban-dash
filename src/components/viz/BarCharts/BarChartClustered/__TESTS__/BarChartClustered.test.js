import React from 'react';
import { shallow } from 'enzyme';
import BarChartClustered from '../index';

const testData = [
  {
    dataSetName: 'first',
    description: 'this is first',
    values: [
      { x: 0, y: 10 },
      { x: 1, y: 20 },
      { x: 2, y: 15 },
      { x: 3, y: 40 },
    ],
  },
  {
    dataSetName: 'second',
    description: 'this is second',
    values: [
      { x: 0, y: 20 },
      { x: 1, y: 30 },
      { x: 2, y: 19 },
      { x: 3, y: 50 },
    ],
  },
];

const testTitle = 'chart title';
const testIcon = 'trending-up';
const testIconSize = 'md';

describe('<BarChartClustered />', () => {
  test('snapshot', () => {
    const wrapper = shallow(<BarChartClustered data={testData} title={testTitle} iconType={testIcon} iconSize={testIconSize} />);
    expect(wrapper).toMatchSnapshot();
  });
});
