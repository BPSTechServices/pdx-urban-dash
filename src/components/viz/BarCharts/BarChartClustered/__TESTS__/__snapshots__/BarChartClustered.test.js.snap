// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarChartClustered /> snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BarChartClustered
    className=""
    colorScheme="light"
    data={
      Array [
        Object {
          "dataSetName": "first",
          "description": "this is first",
          "values": Array [
            Object {
              "x": 0,
              "y": 10,
            },
            Object {
              "x": 1,
              "y": 20,
            },
            Object {
              "x": 2,
              "y": 15,
            },
            Object {
              "x": 3,
              "y": 40,
            },
          ],
        },
        Object {
          "dataSetName": "second",
          "description": "this is second",
          "values": Array [
            Object {
              "x": 0,
              "y": 20,
            },
            Object {
              "x": 1,
              "y": 30,
            },
            Object {
              "x": 2,
              "y": 19,
            },
            Object {
              "x": 3,
              "y": 50,
            },
          ],
        },
      ]
    }
    iconSize=""
    iconType=""
    icon_size="md"
    icon_type="trending-up"
    title="chart title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <a />,
        <Legend
          series={
            Array [
              Object {
                "color": "#82204A",
                "description": "this is first",
                "title": "first",
              },
              Object {
                "color": "#558C8C",
                "description": "this is second",
                "title": "second",
              },
            ]
          }
        />,
      ],
      "className": "",
      "iconSize": "",
      "iconType": "",
      "title": "chart title",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "series": Array [
            Object {
              "color": "#82204A",
              "description": "this is first",
              "title": "first",
            },
            Object {
              "color": "#558C8C",
              "description": "this is second",
              "title": "second",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <a />,
          <Legend
            series={
              Array [
                Object {
                  "color": "#82204A",
                  "description": "this is first",
                  "title": "first",
                },
                Object {
                  "color": "#558C8C",
                  "description": "this is second",
                  "title": "second",
                },
              ]
            }
          />,
        ],
        "className": "",
        "iconSize": "",
        "iconType": "",
        "title": "chart title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "series": Array [
              Object {
                "color": "#82204A",
                "description": "this is first",
                "title": "first",
              },
              Object {
                "color": "#558C8C",
                "description": "this is second",
                "title": "second",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
